lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1953,23 +1957,43 @@ void FIRRTLLowering::initializeRegister(Value reg, Value resetSignal) {
   // Construct and return a new reference to `RANDOM.  It is always a 32-bit
   // unsigned expression.  Calls to $random have side effects, so we use
   // VerbatimExprSEOp.
+  constexpr unsigned RANDOM_WIDTH = 32;
   auto getRandom32Val = [&]() -> Value {
-    return builder.create<sv::VerbatimExprSEOp>(builder.getIntegerType(32),
-                                                "`RANDOM");
+    return builder.create<sv::VerbatimExprSEOp>(
+        builder.getIntegerType(RANDOM_WIDTH), "`RANDOM");
   };
 
   // Return an expression containing random bits of the specified width.
   // An explicit std::function is required here due to recursion.
   std::function<Value(IntegerType)> getRandomValue =
       [&](IntegerType type) -> Value {
     assert(type.getWidth() != 0 && "zero bit width's not supported");
-    auto rand32 = getRandom32Val();
-    if (type.getWidth() <= 32)
-      return builder.createOrFold<comb::ExtractOp>(type, rand32, 0);
+    auto &randomValueAndRemain = blockRandomValueAndRemain[builder.getBlock()];
```

Comment:
Another option might be to have one wire of random, and keep increasing the type as needed (comcatening in new random calls as needed).
```
wire X = {`Rand, `Rand};
use slices of x for all initializations.

Reply:
Oh, it’s a good idea. Increasing the width of wire and replacing uses might be costly therefore l might rather create a super long wire beforehand and shrink the width at post processing.

Reply:
It seems `extract(concat(Rand, Rand, ...., ))` is canonicalized to `concat(extract(Rand), ...)` eventually. So I would like to use as it is.